src/App.tsx

```diff
@@ -12,36 +13,93 @@ import { PdfViewerAPI } from './types/PdfViewerAPI';
-
   const editorRef = React.useRef<EditorAPI>(null);
-  const pdfViewerRef = React.useRef<PdfViewerAPI>(null);
-  const handleReferenceClicked = (reference: ReferenceItem) => {
-    editorRef.current?.insertReference(reference);
-  };
 
+  const pdfViewerRef = React.useRef<PdfViewerAPI>(null);
   const updatePDFViewerWidth = useCallback(() => {
     pdfViewerRef.current?.updateWidth();
   }, [pdfViewerRef]);
 
   return (
-    <PanelGroup autoSaveId="refstudio" className="h-full" direction="horizontal" onLayout={updatePDFViewerWidth}>
-      <PrimarySideBar activePane={primaryPane} onClick={setPrimaryPane} />
-      <Panel collapsible defaultSize={20}>
+    <PanelGroup
+      autoSaveId="refstudio"
+      className="relative h-full"
+      direction="horizontal"
+      onLayout={updatePDFViewerWidth}
+    >
+      <LeftSidePanelWrapper onRefClicked={(reference) => editorRef.current?.insertReference(reference)} />
+      <Panel defaultSize={60}>
+        <MainPanel editorRef={editorRef} pdfViewerRef={pdfViewerRef} />
+      </Panel>
+      <RightPanelWrapper />
+    </PanelGroup>
+  );
+}
+
+function LeftSidePanelWrapper({ onRefClicked }: { onRefClicked(item: ReferenceItem): void }) {
+  const leftPanelRef = React.useRef<ImperativePanelHandle>(null);
+  const [primaryPaneCollapsed, setPrimaryPaneCollapsed] = useState(false);
+  const [primaryPane, setPrimaryPane] = useState<PrimarySideBarPane>('Explorer');
+
+  function handleSideBarClick(selectedPane: PrimarySideBarPane) {
+    if (selectedPane === primaryPane) {
+      // Toggle collapsing
+      setPrimaryPaneCollapsed(!primaryPaneCollapsed);
+    } else {
+      // Always open the sidebar
+      setPrimaryPaneCollapsed(false);
+    }
+    setPrimaryPane(selectedPane);
+  }
+
+  React.useEffect(() => {
+    if (primaryPaneCollapsed) {
+      leftPanelRef.current?.collapse();
+    } else {
+      leftPanelRef.current?.expand();
+    }
+  }, [leftPanelRef, primaryPaneCollapsed]);
+
+  return (
+    <>
+      <PrimarySideBar activePane={primaryPaneCollapsed ? null : primaryPane} onClick={handleSideBarClick} />
+      <Panel
+        collapsible
+        defaultSize={20}
+        ref={leftPanelRef}
+        onCollapse={(collapsed) => setPrimaryPaneCollapsed(collapsed)}
+      >
         {primaryPane === 'Explorer' && <ExplorerPanel />}
-        {primaryPane === 'References' && <ReferencesPanel onRefClicked={handleReferenceClicked} />}
+        {primaryPane === 'References' && <ReferencesPanel onRefClicked={onRefClicked} />}
       </Panel>
       <VerticalResizeHandle />
+    </>
+  );
+}
 
-      <Panel defaultSize={60}>
-        <MainPanel editorRef={editorRef} pdfViewerRef={pdfViewerRef} />
-      </Panel>
+function RightPanelWrapper() {
+  const [closed, setClosed] = React.useState(false);
 
+  if (closed) {
+    return (
+      <div className="absolute bottom-0 right-0 flex border border-slate-300 bg-slate-100 px-4 py-2">
```

Comment:
Just a comment about the UX: it feels a bit weird to me to have to click on the chevron to reopen the panel. It would feel more natural to be able to click anywhere on this div to reopen. What do you think?

Reply:
Yes, perfect! I agree the icon should stay